src/Microsoft.OData.Core/Json/JsonReader.cs

```diff
@@ -2284,16 +2395,226 @@ private void CopyInputToBuffer()
         /// <returns>32-bit signed integer equivalent.</returns>
         private int ParseUnicodeHexValue()
         {
-            string unicodeHexValue = this.ConsumeTokenToString(4);
+            ReadOnlySpan<char> unicodeHexValue = this.ConsumeTokenToSpan(4);
             int characterValue;
             if (!Int32.TryParse(unicodeHexValue, NumberStyles.HexNumber, CultureInfo.InvariantCulture, out characterValue))
             {
-                throw JsonReaderExtensions.CreateException(Error.Format(SRResources.JsonReader_UnrecognizedEscapeSequence, "\\u" + unicodeHexValue));
+                throw JsonReaderExtensions.CreateException(Error.Format(SRResources.JsonReader_UnrecognizedEscapeSequence, "\\u" + unicodeHexValue.ToString()));
             }
 
             return characterValue;
         }
 
+        /// <summary>
+        /// Determines whether the specified span of characters is equal to the expected span of characters.
+        /// </summary>
+        /// <param name="current">The span of characters to compare.</param>
+        /// <param name="expect">The span of characters to compare against.</param>
+        /// <returns><see langword="true"/> if the spans are of equal length and contain the same characters in the same order; otherwise, <see langword="false"/>.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool Is(ReadOnlySpan<char> current, ReadOnlySpan<char> expect) =>
+            current.Length == expect.Length && current.SequenceEqual(expect);
+
+        /// <summary>
+        /// Attempts to match a given span of characters to a predefined set of common OData property names and returns the corresponding interned string if a match is found.
+        /// The method is intended to reduce memory usage by  interning commonly used property names in OData payloads
+        /// </summary>
+        /// <param name="span">The span of characters to evaluate. This span is compared against known OData property names.</param>
+        /// <param name="value">When this method returns, contains the interned string corresponding to the matched OData property name,  if
+        /// the match is successful; otherwise, <see langword="null"/>.</param>
+        /// <returns><see langword="true"/> if the span matches one of the predefined OData property names;  otherwise, <see langword="false"/>.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool TryInternCommonPropertyNames(ReadOnlySpan<char> span, out string value)

```

Comment:
Based on this method name, I thought it was based on [`string.Intern`](https://learn.microsoft.com/en-us/dotnet/fundamentals/runtime-libraries/system-string-intern), but it's not. So to me it's a bit confusing. I'd suggest using a different name, e.g. `TryGetMatchingCommonPropertyNameString`